### ISSUE TITLE
create scripts to get basic stats from pgstatsmon

### ISSUE THREAD

**Issue by KodyKantor** [joyent/pgstatsmon#11]
We can use the node-artedi and pgstatsmon dtrace probes to get some useful information out of pgstatsmon. This will help operators debug when and where things are going wrong with pgstatsmon.

**Comment by KodyKantor** [comment]
CR for this: https://cr.joyent.us/#/c/3936

I wrote three dtrace scripts to help operators: backendstat.d, querystat.d, and walstat.d.

backendstat.d is meant to help operators identify when pgstatsmon is having problems with specific backends. Here is some sample output:
```
$ ./backendstat.d
                                 BACKEND    LAT QTIM QERR CERR  NaN
 3.postgres.walleye.kkantor.com-cd1cbdc3    120    0    0    0    0
 3.postgres.walleye.kkantor.com-8ad69f3e    121    0    0    0    0
 2.postgres.walleye.kkantor.com-a4740e28    122    0    2    0    0
 1.postgres.walleye.kkantor.com-6cc0e780    123    0    2    0    0
 2.postgres.walleye.kkantor.com-64964c42    123    0    2    0    0
 1.postgres.walleye.kkantor.com-36dabbd9    125    0    2    0    0
                                    tick    127
```
From this output we can see that all but two backends are reporting query errors. This is because all but those two backends are running on Postgres 9.2. The queries to retrieve replication statistics only work on Postgres 9.4+. The queries fail on the version 9.2 backends, but are non-fatal failures.

We can also see that the entire 'tick' took 127ms from start to finish. This table prints out ever pgstatsmon 'tick.'

querystat.d can be used to 'drill down' from the results of backendstat.d. If we noticed that one backend was returning bogus metrics we can look at querystat.d to see if there were any problems with the queries. Here's some sample output from one of the above Postgres 9.2 backends (with failing queries):
```
$ ./querystat.d 2.postgres.walleye.kkantor.com-a4740e28
2.postgres.walleye.kkantor.com-a4740e28
               QUERY    LAT QTIM QERR  NaN
 pg_stat_user_tables     29    0    0    0
 pg_stat_replication     79    0    1    0
         pg_recovery     84    0    1    0
    pg_stat_activity     92    0    0    0
    pg_stat_database    105    0    0    0
    pg_relation_size    119    0    0    0
    pg_stat_bgwriter    133    0    0    0
           pg_vacuum    144    0    0    0
```

The table shows that the pg_stat_replication and pg_recovery queries fail. If we look at the logs we can see why (the `pg_lsn` type doesn't exist until 9.4). We can also see some latency stats. These are cumulative latencies, which can be a little confusing. node-postgres queues queries that stack up on a given connection. So pgstatsmon thinks multiple queries are in flight for each backend, but in reality they're sitting in a node-postgres queue racking up latency. This table gets printed on every pgstatsmon 'tick.'

walstat.d shows WAL throughput information in a similar format to Manatee's `manatee-adm pg-status`. Here's example output:
```
     ----- CLUSTER STATE -----      ----- THROUGHPUT (Bytes/tick) -----
     SHARD       ZONE       REPL       SENT    WRITTEN    FLUSHED   REPLAYED
         3   cd1cbdc3       sync      64496      64496      64496      64496
```

Hopefully the headers here are descriptive enough to understand what's going on. The `SENT` column indicates the number of bytes sent to the downstream peer. `WRITTEN`, `FLUSHED`, and `REPLAYED` are all based on data points reported by the downstream peer. This table is printed on every pgstatsmon 'tick.' This script is really tailored for Manta's Manatee. There is one line per backend. Again, this example only has one backend because only one shard runs PG 9.6, and these stats are only available on non-recovering peers (primaries).
